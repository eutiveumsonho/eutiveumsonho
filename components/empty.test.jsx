import { render, screen } from "@testing-library/react";
import Empty from "./empty";
import "jest-styled-components";

jest.mock("next/router", () => ({
  useRouter: () => ({
    push: () => null,
  }),
}));

describe("Empty", () => {
  const empty = {
    description: "Test description",
    label: "Action",
    actionRoute: "/",
  };

  it("render props", () => {
    render(<Empty empty={empty} />);

    const description = screen.getByText(empty.description);
    const label = screen.getByText(empty.label);

    expect(description).toBeInTheDocument();
    expect(label).toBeInTheDocument();
  });

  it("renders empty component unchanged", () => {
    const { container } = render(<Empty empty={empty} />);
    expect(container).toMatchSnapshot();
  });
});
