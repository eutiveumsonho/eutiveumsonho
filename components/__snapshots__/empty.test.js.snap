// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty renders empty component unchanged 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 96px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  height: 12px;
}

.c1 {
  font-size: 18px;
  line-height: 24px;
}

.c4 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background-color: #7D4CDB;
  color: #f8f8f8;
  border-radius: 18px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

.c4:hover {
  box-shadow: 0px 0px 0px 2px #7D4CDB;
}

.c4:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c4:focus > circle,
.c4:focus > ellipse,
.c4:focus > line,
.c4:focus > path,
.c4:focus > polygon,
.c4:focus > polyline,
.c4:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c4:focus::-moz-focus-inner {
  border: 0;
}

.c4:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c4:focus:not(:focus-visible) > circle,.c4:focus:not(:focus-visible) > ellipse,
.c4:focus:not(:focus-visible) > line,.c4:focus:not(:focus-visible) > path,
.c4:focus:not(:focus-visible) > polygon,.c4:focus:not(:focus-visible) > polyline,
.c4:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c4:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

@media only screen and (max-width:768px) {
  .c0 {
    padding: 48px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    height: 6px;
  }
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1"
      style="text-align: center;"
    >
      Test description
    </span>
    <div
      class="c2"
    />
    <div
      class="c3"
    >
      <button
        class="c4"
        type="button"
      >
        Action
      </button>
    </div>
  </div>
</div>
`;
